processing/src/main/java/io/druid/query/CPUTimeMetricQueryRunner.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.metamx.common.ISE;
+import com.metamx.common.guava.Accumulator;
+import com.metamx.common.guava.Sequence;
+import com.metamx.common.guava.Sequences;
+import com.metamx.common.guava.Yielder;
+import com.metamx.common.guava.YieldingAccumulator;
+import com.metamx.emitter.service.ServiceEmitter;
+import com.metamx.emitter.service.ServiceMetricEvent;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class CPUTimeMetricQueryRunner<T> implements QueryRunner<T>
+{
+  private static final ThreadMXBean THREAD_MX_BEAN = ManagementFactory.getThreadMXBean();
+  private final QueryRunner<T> delegate;
+  private final Function<Query<T>, ServiceMetricEvent.Builder> builderFn;
+  private final ServiceEmitter emitter;
+  private final AtomicLong cpuTimeAccumulator;
+  private final boolean report;
+
+  private CPUTimeMetricQueryRunner(
+      QueryRunner<T> delegate,
+      Function<Query<T>, ServiceMetricEvent.Builder> builderFn,
+      ServiceEmitter emitter,
+      AtomicLong cpuTimeAccumulator,
+      boolean report
+  )
+  {
+    if (!THREAD_MX_BEAN.isThreadCpuTimeEnabled()) {
+      throw new ISE("Cpu time must enabled");
```

Comment:
documentation is not very explicit about when this will happen. Is it possible to not fail here and print some warn/error log saying cpu metrics can't be reported?